pyxem/generators/red_intensity_generator.py

```diff
@@ -0,0 +1,213 @@
+
+        Parameters
+        ----------
+        calibration: float
+                    Calibration in terms of A^-1 per pixel.
+        """
+        self.signal.axes_manager.signal_axes[0].scale = calibration
+        return
+
+    def specify_cutoff_vector(self, s_min, s_max):
+        """
+        Scattering vector cutoff for the purposes of fitting an atomic scattering
+        factor to the 1D profile. Specified in terms of s (in inverse angstroms).
+        s is defined as s = 2 sin(theta)/lambda = 1/d.
+
+        Parameters
+        ----------
+        s_min: float
+                    Minimum scattering vector amplitude for cutoff.
+        s_max: float
+                    Maximum scattering vector amplitude for cutoff.
+        """
+        #s_scale = self.signal.axes_manager.signal_axes[0].scale
+        self.cutoff = [s_min, s_max]
+        return
+
+    def fit_atomic_scattering(self, elements, fracs,
+                              N=1., C=0., type='lobato',
+                              plot_fit=True):
+        """Fits a diffraction intensity profile to the background using
+        FIT = N * sum(ci * (fi^2) + C)
+
+        The cutoff for the scattering factor fit to s is defined via the function
+        specify_cutoff_vector above.
+
+        Parameters
+        ----------
+        elements: list of str
+                    A list of elements present (by symbol).
+        fracs: list of float
+                    A list of fraction of the respective elements. Should sum to 1.
+        N : float
+                    The "slope" of the fit.
+        C : float
+                    An additive constant to the fit.
+        type : str
+                    Type of scattering parameters fitted. Default is lobato.
+                    See scattering_fit_component for more details.
+        plot_fit: bool
+                    A bool to decide if the fit from scattering is plotted
+                    after fitting.
+        """
+
+        fit_model = self.signal.create_model()
+        background = ScatteringFitComponent(elements, fracs, N, C, type)
+
+        fit_model.append(background)
+        fit_model.set_signal_range(self.cutoff)
+        fit_model.multifit()
+        fit_model.reset_signal_range()
+        if plot_fit == True:
+            fit_model.plot()
+        C_values = background.C.as_signal()
+        N_values = background.N.as_signal()
+        s_size = self.sig_size[0]
+        s_scale = self.signal.axes_manager.signal_axes[0].scale
+        fit, normalisation = scattering_to_signal(elements, fracs, N_values,
+                                                  C_values, s_size, s_scale, type)
+        # self.fit = np.array(background.sum_squares).reshape(
+        #            self.nav_size[0],self.nav_size[1],self.sig_size[0])
+
+        self.normalisation = normalisation  # change this
+        self.background_fit = fit
+        return
+
+    def subtract_bkgd_pattern(self, bkgd_pattern):
```

Comment:
Hmmm, I'm not sure about this behaviour.... So I'm writing a comment to think about it. @pc494 , what do you think here?

Reply:
I don't love it, at very least it should have `return None`, I think I would prefer it as a map?

Reply:
Changed to map.